core/src/main/java/org/elasticsearch/indices/store/IndicesStore.java

```diff
@@ -146,27 +185,12 @@ boolean shardCanBeDeleted(ClusterState state, IndexShardRoutingTable indexShardR
 
         for (ShardRouting shardRouting : indexShardRoutingTable) {
             // be conservative here, check on started, not even active
-            if (!shardRouting.started()) {
+            if (shardRouting.started() == false) {
                 return false;
             }
 
-            // if the allocated or relocation node id doesn't exists in the cluster state  it may be a stale node,
-            // make sure we don't do anything with this until the routing table has properly been rerouted to reflect
-            // the fact that the node does not exists
-            DiscoveryNode node = state.nodes().get(shardRouting.currentNodeId());
```

Comment:
it's a happy day we can remove this and because our nodes and routing table is consistent. We should probably change this in deleteShardIfExistElseWhere:

```
            // Node can't be null, because otherwise shardCanBeDeleted() would have returned false
            DiscoveryNode currentNode = state.nodes().get(shardRouting.currentNodeId());
            assert currentNode != null;
```

Also - do we have assertions for this ClusterState builder, so we're sure it never happens?